backend/src/panels/GuidedDevelopmentPanel.ts

```diff
@@ -15,17 +15,45 @@ export class GuidedDevelopmentPanel extends AbstractWebviewPanel {
 	public static GUIDED_DEVELOPMENT = "Guided Development";
 
 	private static channel: vscode.OutputChannel;
+	private uiOptions: any;
 
 	public toggleOutput() {
 		this.outputChannel.showOutput();
 	}
 
+	public loadWebviewPanel(uiOptions?: any) {
+		if (this.webViewPanel && JSON.stringify(uiOptions)===JSON.stringify(this.uiOptions)) {
```

Comment:
Depending on specific scenario here, the `JSON.stringify` strategy might work, however, there might be other concerns in extreme cases? E.g., one of which, it doesn't guarantee the order of properties, so:
```JavaScript
JSON.stringify({title: 'title', description: 'description'}) 
    !== 
JSON.stringify({description: 'description', title: 'title'})
```
What about considering the `isEqual` in lodash which has been available in this extension?

Reply:
For isEqual in lodash, Object objects are compared by their own, not inherited, enumerable properties. Functions and DOM nodes are not supported, so _.isEqual({a: 'a', b:function(){return "b1"}}, {a: 'a', b: function(){return "b1"}}) will return false not as expected.  
As uiOptions is generated in codes with certain order of properties, I use JSON.stringify currently.

Reply:
> For isEqual in lodash, Object objects are compared by their own, not inherited, enumerable properties. Functions and DOM nodes are not supported, so _.isEqual({a: 'a', b:function(){return "b1"}}, {a: 'a', b: function(){return "b1"}}) will return false not as expected. As uiOptions is generated in codes with certain order of properties, I use JSON.stringify currently.

Yes I agree it fits in the current scenario. As we discussed, `JSON.stringify` also has similar behaviors in terms of inherited properties. I agree to make it progressively especially for functions as you mentioned, whether reference-level equality, or AST-level equality and others.